pfioh/pfioh.py

```diff
@@ -818,63 +804,56 @@ def do_POST_withCompression(self, **kwargs):
-                with open(str_localFile, 'wb') as fh:
-                    fh.write(data)
+                data                        = base64.b64decode(fileContent)
                 d_ret['decode']['status']   = True
                 d_ret['decode']['msg']      = 'base64 decode successful!'
-            except:
+            except Exception as err:
                 d_ret['decode']['status']   = False
                 d_ret['decode']['msg']      = 'base64 decode unsuccessful!'
                 self.ret_client(d_ret)
                 self.qprint(d_ret, comms = 'tx')
+                self.qprint(err)
                 return d_ret
-            d_ret['decode']['timestamp']  = '%s' % datetime.datetime.now()
+
         else:
             d_ret['write']   = {}
-            with open(str_localFile, 'wb') as fh:
-                try:
-                    fh.write(fileContent.decode())
-                    d_ret['write']['decode'] = True
-                except:
-                    fh.write(fileContent)
-                    d_ret['write']['decode'] = False
-            d_ret['write']['file']      = str_localFile
-            d_ret['write']['status']    = True
-            d_ret['write']['msg']       = 'File written successfully!'
-            d_ret['write']['filesize']  = "{:,}".format(os.stat(str_localFile).st_size)
-            d_ret['write']['timestamp'] = '%s' % datetime.datetime.now()
-            d_ret['status']             = True
-            d_ret['msg']                = d_ret['write']['msg']
-        fh.close()
-        if b_unpack and d_compress['archive'] == 'zip':
-            d_fio   =   zip_process(action          = 'unzip',
-                                    path            = str_unpackPath,
-                                    payloadFile     = str_localFile)
-            d_ret['unzip']  = d_fio
-            d_ret['status'] = d_fio['status']
-            d_ret['msg']    = d_fio['msg']
-            os.remove(str_localFile)
-
+            try:
+                data = fileContent.decode()
+                d_ret['write']['decode'] = True
+            except Exception as err:
+                d_ret['write']['decode'] = False
+                data = fileContent
+                self.qprint(err)
+
+        b_zip= b_unpack and (d_compress['archive'] == 'zip')
+
+        d_ret= self.storeData(file_name= str_localFile, 
+                              file_content= data, Path= str_unpackPath, is_zip=b_zip,d_ret=d_ret)
+        
         # pudb.set_trace()
         d_ret['postop'] = self.postop_process(meta          = d_meta,
                                               path          = str_unpackPath,
                                               payloadFile   = str_localFile)
 
+
         self.send_response(200)
         self.end_headers()
 
         d_ret['User-agent'] = self.headers['user-agent']
 
         # self.ret_client(d_ret)
         self.qprint(d_ret, comms = 'tx')
-
         return d_ret
 
+    def storeData(self, **kwargs):
```

Comment:
Nit:

> it will first search for the method in the class whose object I have used to call the method with

Technically I think you mean it will make the call from the object that was instantiated.  Which might not be the class that is making the call.

I was really just asking if it was required in some way.  Didn't know why you would have done it in the first place unless it was a pattern you were familiar with.  I can't say I love the pattern as it would already give an error if it wasn't implemented.  And it just adds cruft to the code.  But not being a long time python coder, I can't comment on whether it's idiomatic.